libpromises/addr_lib.c

```diff
@@ -578,23 +578,42 @@ void ParseHostPort(char *s, char **hostname, char **port)
 {
     char *h, *p;                              /* hostname, port temporaries */
 
-    if (s[0] == '[')                           /* IPv6 form: [address]:port */
+    h = s;
+    p = NULL;
+
+    char *first_colon = strchr(s, ':');
+    char *first_dot   = strchr(s, '.');
+
+    if(s[0] == '[')                         // [host or ip]:port
     {
         h = s + 1;
         p = strchr(h, ']');
         if (p != NULL)
         {
-            *p = '\0';                           /* '\0'-terminate hostname */
-            p = (p[1] == ':') ? p+2 : NULL;
+            *p = '\0';                      // '\0' terminate host name
+            p = (p[1] == ':') ? p+2 : NULL; // move port* forward or set NULL
+        }
+    }
+    else if( first_colon == NULL )          // no colon, no port, host correct
+    {
+        // Do nothing
+    }
+    else if( first_colon < first_dot || first_dot == NULL)
```

Comment:
Please check first for `first_dot == NULL` in this expression, to avoid further examining `first_dot` if that's the case. It shouldn't change anything in functionality.